tests/smoke_test/test_basic.py

```diff
@@ -83,4 +83,4 @@ def test_kernel_not_tainted(shell):
     """Check if the kernel is not tainted - do it at the end of the
     test suite to increase the chance of catching issues."""
     output = shell.run_check("cat /proc/sys/kernel/tainted")
-    assert output == "0\n", f"Kernel tainted: {output}"
+    assert "\n".join(output) == "0", f"Kernel tainted: {output}"
```

Comment:
Isn't that fixing it on both sides? :thinking: 

How about:
```suggestion
    assert output.strip() == "0", f"Kernel tainted: {output}"
```

Reply:
That won't work, the thing is the return type is a list of strings. This time I actually checked the test doesn't fail :see_no_evil:

Reply:
Hm, I see, and `"\n".join(output)` won't add a newline if there is only one element. Yeah make sense.